### ISSUE TITLE
zero AP

### ISSUE THREAD

**Issue by enhany** [DeNA/PyTorch_YOLOv3#55]
I train with custom dataset (1k images / 1 class) with default config and get zero AP, but loss is quit low.
The same dataset in another models works well.

```
[Iter 7950/50000] [lr 0.001000] [Losses: xy 0.002690, wh 0.008716, conf 2.793764, cls 0.002621, total 1.024159, imgsize 352]
[Iter 7960/50000] [lr 0.001000] [Losses: xy 0.001360, wh 0.000597, conf 2.744817, cls 0.001324, total 0.983985, imgsize 320]
[Iter 7970/50000] [lr 0.001000] [Losses: xy 0.002676, wh 0.005998, conf 2.818687, cls 0.002608, total 1.026828, imgsize 608]
[Iter 7980/50000] [lr 0.001000] [Losses: xy 0.005337, wh 0.004117, conf 2.839865, cls 0.005201, total 1.035145, imgsize 576]
[Iter 7990/50000] [lr 0.001000] [Losses: xy 0.004003, wh 0.001313, conf 2.766982, cls 0.003901, total 0.993268, imgsize 544]
Loading and preparing results...
DONE (t=0.01s)
creating index...
index created!
Running per image evaluation...
Evaluate annotation type *bbox*
DONE (t=0.10s).
Accumulating evaluation results...
DONE (t=0.02s).
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.000
 Average Precision  (AP) @[ IoU=0.50      | area=   all | maxDets=100 ] = 0.000
 Average Precision  (AP) @[ IoU=0.75      | area=   all | maxDets=100 ] = 0.000
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.000
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.000
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=  1 ] = 0.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets= 10 ] = 0.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.000
```

What direction I have to look at to fix it?

**Comment by hirotomusiker** [comment]
Thank you, but unfortunately we have no idea how to fix the issue because we do not have your dataloader.
Could you check if the data returned form your dataloader are in the same format as COCO's.
see : https://github.com/DeNA/PyTorch_YOLOv3/blob/master/utils/utils.py#L187-L191
